AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java

```diff
@@ -79,13 +80,24 @@ protected void processParameters(HasPreferences holder, JMethod listenerMethod,
 		JVar newValueParam = listenerMethod.param(classes().OBJECT, "newValue");
 
 		for (VariableElement variableElement : userParameters) {
-			if (variableElement.asType().toString().equals(CanonicalNameConstants.PREFERENCE)) {
+			String type = variableElement.asType().toString();
+
+			if (type.equals(CanonicalNameConstants.PREFERENCE)) {
 				call.arg(preferenceParam);
-			} else if (variableElement.asType().toString().equals(CanonicalNameConstants.OBJECT)) {
+			} else if (type.equals(CanonicalNameConstants.OBJECT)) {
 				call.arg(newValueParam);
+			} else if (type.equals(CanonicalNameConstants.INTEGER) || type.equals(int.class.getName()) || //
+					type.equals(CanonicalNameConstants.FLOAT) || type.equals(float.class.getName()) || //
+					type.equals(CanonicalNameConstants.LONG) || type.equals(long.class.getName())) {
+				JClass wrapperClass = type.startsWith("java") ? holder.refClass(type) : JType.parse(holder.codeModel(), type.replace(".class", "")).boxify();
```

Comment:
This is quite hard to understand. Is there another way to get the wrapper class ?